src/net/autoupdate.cpp

```diff
@@ -108,6 +108,7 @@ AutoUpdater::VersionInfo AutoUpdater::getUpdateVersion()
 
     QNetworkAccessManager *manager = new QNetworkAccessManager;
     QNetworkReply* reply = manager->get(QNetworkRequest(QUrl(checkURI)));
+    manager->setProxy(Settings::getInstance().getProxy());
```

Comment:
No! 
manager->get() will immediately start the request, at best you're racing to set the proxy, but it'll probably just not be proxied at all.